### DISCUSSION TITLE
How to use aws-iot-device-sdk-cpp-v2 with CPM?

### DISCUSSION THREAD

**Discussion by mateuszboryn** [aws/aws-iot-device-sdk-cpp-v2#752] [General]
Hi, please let me know if somebody used aws-iot-device-sdk-cpp-v2 with CPM? https://github.com/cpm-cmake/CPM.cmake
As I read the docs for aws-iot-device-sdk-cpp-v2 it requires `make install` even before building the samples. The samples assume the lib is already built.
What is the best way to invoke CPM, referencing a released version of aws-iot-device-sdk-cpp-v2 and link that to some target in my project?

What I need is fully automatic way of downloading and building aws-iot-device-sdk-cpp-v2 as one of many many dependencies in my project. I don't want to do some manual actions before building my project, like separately downloading aws-iot-device-sdk-cpp-v2 and building it by hand.
I need it to be built similar way to other CPM examples, like this one: https://github.com/cpm-cmake/CPM.cmake/blob/master/examples/boost/CMakeLists.txt#L14

Doing just this
```
CPMAddPackage(
    NAME aws-iot-device-sdk-cpp-v2
    GITHUB_REPOSITORY aws/aws-iot-device-sdk-cpp-v2
    VERSION 1.33.0
    GIT_TAG 7a1a841e6fbe54d7516a8b26448de358a4980c0e
    OPTIONS "USE_OPENSSL ON"
)
```
is not enough, because:
- CPM does only git clone --recursive, and configure cmake
- CPM does not build targets at my project's confiugre time
- To build anything that depends on aws-iot-device-sdk-cpp-v2, it must be fully built (so that aws-iot-device-sdk-cpp-v2 deps are present and aws-iot-device-sdk-cpp-v2 itself is present), only after that I could start configuring my project

What I'm looking for is replacing these 2 lines
https://github.com/aws/aws-iot-device-sdk-cpp-v2/blob/main/samples/fleet_provisioning/mqtt5_fleet_provisioning/CMakeLists.txt#L23-L24

with a reference to outputs of `CPMAddPackage(NAME aws-iot-device-sdk-cpp-v2 ...` mentioned above in this description.

**Comment by jmklix** [comment]
This is not something that we currently support

**Comment by paulocoutinhox** [comment]
Hi,

I have the same problem here:
https://github.com/aws/aws-sdk-cpp/discussions/3301

Did you find any solution?

Thanks.
